### PR TITLE
h3: support SETTINGS_ENABLE_CONNECT_PROTOCOL

### PR DESCRIPTION
**Pull request by ehaydenr**

### CODE REVIEW

File: quiche/src/h3/mod.rs [hunk 1]
```diff
@@ -552,6 +554,13 @@ impl Config {
     pub fn set_qpack_blocked_streams(&mut self, v: u64) {
         self.qpack_blocked_streams = Some(v);
     }
+
+    /// Sets or omits the `SETTINGS_ENABLE_CONNECT_PROTOCOL` setting.
+    ///
+    /// The default value is `0`.
+    pub fn set_connect_protocol_enabled(&mut self, v: u64) {
```
Comment (LPardue) [inline quiche/src/h3/mod.rs:561]: this would probably be better modelled as a bool such as `enable_extended_connect(enabled: bool)`

Comment (LPardue) [reply quiche/src/h3/mod.rs:561]: modelling the setting internally as an uint is fine, but there's little reason for us to let users of quiche set random values when the RFC doesn't permit that

Comment (ehaydenr) [reply quiche/src/h3/mod.rs]: thanks, addressed in ced7972025b1c0993642df69fa09520713d320bc

File: quiche/src/h3/mod.rs [hunk 2]
```diff
@@ -1280,6 +1296,17 @@ impl Connection {
...
     }
 
+    /// Returns whether the peer enabled HTTP/3 CONNECT protocol support.
+    ///
+    /// Support is signalled by the peer's SETTINGS, so this method always
+    /// returns false until they have been processed using the [`poll()`]
+    /// method.
+    ///
+    /// [`poll()`]: struct.Connection.html#method.poll
+    pub fn connect_protocol_enabled_by_peer(&self) -> bool {
```
Comment (LPardue) [inline quiche/src/h3/mod.rs:1306]: for consistency with `enable_extended_connect`, how about 
```suggestion
    pub fn extended_connect_enabled_by_peer(&self) -> bool {
```

File: quiche/src/h3/mod.rs [hunk 3]
```diff
@@ -1280,6 +1296,17 @@ impl Connection {
             conn.dgram_max_writable_len().is_some()
     }
 
+    /// Returns whether the peer enabled HTTP/3 CONNECT protocol support.
```
Comment (LPardue) [inline quiche/src/h3/mod.rs:1299]: for consistency with enable_extended_connect, how about
```suggestion
    /// Returns whether the peer enabled extended CONNECT support.
```

**Review by LPardue — COMMENTED** [review]
This is looking good. For completeness, we probable want to add equivalent capability to the C FFI, meaning adding `quiche_h3_config_enable_extended_connect` to https://github.com/cloudflare/quiche/blob/master/quiche/src/h3/ffi.rs#L72 and https://github.com/cloudflare/quiche/blob/master/quiche/include/quiche.h#L771

**Review by LPardue — APPROVED** [review]

**Review by junhochoi — APPROVED** [review]

**Pull request merged**